decidim-module-census_connector/app/forms/decidim/census_connector/verifications/census/data_handler.rb

```diff
@@ -8,7 +8,7 @@ class DataHandler < CensusHandler
           mimic :data_handler
 
           def self.document_scopes
-            @document_scopes ||= Decidim::Scope.top_level.order(name: :asc)
+            Decidim::Scope.top_level.order(name: :asc)
```

Comment:
I understand that this is to avoid problems when top level scopes are changed, I'm right?

Seeing this, I realize of a mistake that I've done before: scopes queries should include the current organization to avoid mixing scopes from different organizations. I'd change all `Decidim::Scope` for querying with something like `current_organization.scopes`. I don't know if the context includes `current_organization`, or if we should include it in the form context.

Finally, maybe we could consider proposing to Decidim some changes in [the `top_level` method](https://github.com/decidim/decidim/blob/master/decidim-core/app/models/decidim/scope.rb#L45) to avoid repeating this failure in the future. We could modify it to force the inclusion of the organization filtering, for example with an `organization` mandatory parameter. Thoughts?

Reply:
As per the reason for the flaky, yes, that's the problem. We shouldn't memoize at that level.

As per the problem you detected and the proposed fix, I agree. Should I update this PR or wait for the upstream patch? This bug won't bite us anyways since we are not multitenant, right?

Multitenancy is tough...

Reply:
Yes, I'd add it to this PR. It will only bite us if we add another tenant, but I'm sure that solving it now worth it.

Reply:
Ok, will do!